nixos/modules/services/web-apps/nextcloud.nix

```diff
@@ -653,6 +613,41 @@ in {
         freeformType = jsonFormat.type;
         options = {
 
+          dbtype = mkOption {
+            type = types.enum [ "sqlite" "pgsql" "mysql" ];
+            default = "sqlite";
+            description = lib.mdDoc "Database type.";
+          };
+          dbname = mkOption {
+            type = types.nullOr types.str;
+            default = "nextcloud";
+            description = lib.mdDoc "Database name.";
+          };
+          dbuser = mkOption {
+            type = types.nullOr types.str;
+            default = "nextcloud";
+            description = lib.mdDoc "Database user.";
+          };
+          dbhost = mkOption {
+            type = types.nullOr types.str;
+            default =
+              if pgsqlLocal then "/run/postgresql"
+              else if mysqlLocal then "localhost:/run/mysqld/mysqld.sock"
+              else "localhost";
+            defaultText = "localhost";
+            example = "localhost:5000";
+            description = lib.mdDoc ''
+              Database host (+port) or socket path. Defaults to the correct unix socket
+              instead if `services.nextcloud.database.createLocally` is true and
+              `services.nextcloud.extraOptions.dbtype` is either `pgsql` or
+              `mysql`.
+            '';
+          };
+          dbtableprefix = mkOption {
+            type = types.nullOr types.str;
+            default = "oc_";
```

Comment:
I think it's wrong to allow null here and then set the upstream default anyways: I think we should either adhere to upstream defaults wherever possible or use `null`. I'd prefer the former.